test/services/news/index.test.js

```diff
@@ -458,6 +492,276 @@ describe('news service', () => {
+					},
+				]);
+
+				const params = await generateRequestParamsFromUser(user);
+				params.query = { unpublished: true };
+				const result = await newsService.find(params);
+
+				expect(result.total).to.equal(3);
+				expect(
+					result.data.every((item) => item.title.includes('Unpublished')),
+				).to.equal(true);
+			});
+
+			it('should not return unpublished news if unauthorized', async () => {
+				const schoolId = (await createTestSchool())._id;
+				const user = await createTestUser({
+					schoolId,
+					roles: 'student',
+				});
+
+				const user2 = await createTestUser({
+					schoolId,
+					roles: 'teacher',
+				});
+
+				const now = Date.now();
+				await News.create([
+					{
+						schoolId,
+						creatorId: user2._id,
+						title: 'school A news',
+						content: 'some string',
+						displayAt: Date.now(),
+					},
+					{
+						schoolId,
+						creatorId: user2._id,
+						title: 'Unpublished newstitle',
+						content: 'Who know what the future holds',
+						displayAt: now + 20000,
+					},
+					{
+						schoolId,
+						creatorId: user2._id,
+						title: 'Unpublished newstitle (2)',
+						content: 'Unkown event about to happen',
+						displayAt: new Date(now + 20000),
+					},
+					{
+						schoolId,
+						creatorId: user2._id,
+						title: 'Unpublished newstitle (3)',
+						content: 'Future event',
+						displayAt: new Date(now + 80000),
+					},
+					{
+						schoolId,
+						creatorId: user2._id,
+						title: 'school A news (2)',
+						content: 'some string',
+						displayAt: new Date('2019-09-01T10:01:58.135Z'),
+					},
+				]);
+
+				const params = await generateRequestParamsFromUser(user);
+				params.query = { unpublished: true };
+				const result = await newsService.find(params);
+
+				expect(result.total).to.equal(0);
+			});
+
+			it('should be able to sort unpublished news by publish date', async () => {
+				const schoolId = (await createTestSchool())._id;
+				const user = await createTestUser({
+					schoolId,
+					roles: 'teacher',
+				});
+				const now = Date.now();
+				await News.create([
+					{
+						schoolId,
+						creatorId: user._id,
+						title: '1',
+						content: 'this is the content',
+						displayAt: now + 1000 * 30,
+					},
+					{
+						schoolId,
+						creatorId: user._id,
+						title: '2',
+						content: 'even more content',
+						displayAt: new Date(now + 1000 * 20),
+					},
+					{
+						schoolId,
+						creatorId: user._id,
+						title: '3',
+						content: 'content galore',
+						displayAt: new Date(now + 1000 * 80),
+					},
+				]);
+
+				const params = await generateRequestParamsFromUser(user);
+				params.query = { unpublished: true, sort: 'displayAt' };
+
+				const result = await newsService.find(params);
+
+				expect(result.total).to.equal(3);
+				expect(result.data[0].title).to.equal('2');
+				expect(result.data[1].title).to.equal('1');
+				expect(result.data[2].title).to.equal('3');
```

Comment:
test always both ways of sorting, so -displayAt in our case, only then u know, that it is no pure chance in which the items are ordered.

Reply:
done